[TSK-695] [TSK-696] [TSK-697] migration(ui-kit): Available Coaching Session component

ObitoDevv opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: ` Available Coaching Session component`
- Figma link or screenshot: [Figma Design](https://www.figma.com/)
![image](https://github.com/user-attachments/assets/375fbc4d-7ee6-4a88-8ce8-864f9aa343fe)

---

## Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.

---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #44

alebg reviewed (changes requested):
Thanks @ObitoDevv ! ~One general requirement, please expose a place in the storybook controls to manipulate `availableCoachingSessionsData`, so we can play with the session data titles, minutes, etc.~

![immagine](https://github.com/user-attachments/assets/58b285f3-ce2f-4f6f-92ea-810d4d1da7e8)

Apart from that, @carlapbg will also review this PR as usual

ObitoDevv commented:
@alebg I did sir. Can you please verify that everything looks fine ? 

Thank you sir for reviewing !

alebg commented:
@ObitoDevv general question: why am I seeing changes to components, such as Button and UserAvatar, in PRs that don't have to do with those? For example, this PR

alebg reviewed (changes requested):
@ObitoDevv thanks for the changes! 

Are we supposed to be able to use the drag and drop in the storybook? it's not working for me on Firefox

Also, a general question: why am I seeing changes to components, such as Button and UserAvatar, in PRs that don't have to do with those? For example, this PR

Finally, please also address @carlapbg 's review when her's is ready

carlapbg commented:
Great job, thank you. Just one question: This component will be used next to the calendar, so it will appear shorter and not in full screen. Will the component adapt automatically depending on the calendar's size?

<img width="972" alt="Screenshot 2025-03-13 at 00 09 02" src="https://github.com/user-attachments/assets/db50c81c-525e-446c-aef3-2141eb825c6d" />

ObitoDevv commented:
> Great job, thank you. Just one question: This component will be used next to the calendar, so it will appear shorter and not in full screen. Will the component adapt automatically depending on the calendar's size?
> 
> <img alt="Screenshot 2025-03-13 at 00 09 02" width="972" src="https://private-user-images.githubusercontent.com/74186806/422111524-db50c81c-525e-446c-aef3-2141eb825c6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE4NTM5NDksIm5iZiI6MTc0MTg1MzY0OSwicGF0aCI6Ii83NDE4NjgwNi80MjIxMTE1MjQtZGI1MGM4MWMtNTI1ZS00NDZjLWFlZjMtMjE0MWViODI1YzZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEzVDA4MTQwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjOTFmMWRmNDgwYWI3NWI5YjNmMjkzN2JmN2YzZTI0YWFmMDhlNDc5ODY4NzYwNDI3ODhkNDFkMjQyY2YzNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.InafXsuHXVLwpQCNNVO0_OoOojqBaElnzT4AITxT1Xc">

@carlapbg  Yes , this component is responsive for calender's page. It will automatically adapt.

carlapbg reviewed (dismissed):
Looks good to me. Nice job, thank you!

ObitoDevv commented:
> @ObitoDevv thanks for the changes!
> 
> Are we supposed to be able to use the drag and drop in the storybook? it's not working for me on Firefox
> 
> Also, a general question: why am I seeing changes to components, such as Button and UserAvatar, in PRs that don't have to do with those? For example, this PR
> 
> Finally, please also address @carlapbg 's review when her's is ready

@alebg No sir, we haven't added drag and drop functionality here. To implement it, we would need to use dnd-core-kit. For the calendar page, we plan to add dnd-core-kit later. Would you like us to include it here as well? Please confirm.

Regarding the changes to UserAvatar, we have modified it across multiple branches, but this shouldn't cause conflicts with the main branch. To properly handle this, we can create a separate PR with the correct version of UserAvatar and resolve all related conflicts. That’s the main goal behind these changes.

Also, I have addressed @carlapbg's mam issues.

alebg commented on packages/ui-kit/lib/components/available-coaching-sessions/drag-drop-session.tsx:
```diff
@@ -0,0 +1,67 @@
+
+export interface DragDropSessionProps {
+  title?: string;
+  duration?: string;
```
duration should be a number, to represent minutes. Then the component should render the correct word "hours", "minutes", etc based on the localization

alebg commented:
- Buy more sessions button callback
- Unnecessary isEmpty prop

alebg reviewed (dismissed):
thanks @manishRise11 , great job!
